### PR TITLE
[fw-upgrade] make strings translatable #35

### LINKED ISSUE #35: [fw-upgrade] Flag logging strings as translatable
The strings used to store log messages in the upgrader class should be translatable, because these lines are then shown to the user.
Look for `self.log(`: https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/openwisp_firmware_upgrader/upgraders/openwrt.py

See the django docs regarding translatable strings: https://docs.djangoproject.com/en/3.0/topics/i18n/translation/#standard-translation

### PR DESCRIPTION
**Pull request by priyanshuone6**
The strings used to store log messages in the upgrader class is made translatable using the gettext() or _

Closes #35

### CODE REVIEW

File: openwisp_firmware_upgrader/upgraders/openwrt.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 from billiard import Process
 from paramiko.ssh_exception import NoValidConnectionsError
+from django.utils.translation import gettext as _
```
Comment (pandafy) [inline openwisp_firmware_upgrader/upgraders/openwrt.py:8]:
```suggestion
from django.utils.translation import gettext_lazy as _
```

File: openwisp_firmware_upgrader/upgraders/openwrt.py [hunk 2]
```diff
@@ -90,32 +91,32 @@ def _test_checksum(self, image):
                     'Identical checksum found in the filesystem, '
                     'upgrade not needed.'
                 )
-                self.log(message)
+                self.log(_(message))
```
Comment (pandafy) [inline openwisp_firmware_upgrader/upgraders/openwrt.py:94]: Let's translate the string where it is defined instead for better code readability and keep it as it is


   
```suggestion
                 message = _(
                    'Firmware already upgraded previously. '
                    'Identical checksum found in the filesystem, '
                    'upgrade not needed.'
                )
                self.log(message)
```

File: openwisp_firmware_upgrader/upgraders/openwrt.py [hunk 3]
```diff
@@ -160,25 +161,25 @@ def _write_checksum(self, checksum):
...
-                self.log(
+                self.log(_(
                     'Device not reachable yet, '
                     f'retrying in {self.RECONNECT_RETRY_DELAY} seconds...'
-                )
+                ))
                 sleep(self.RECONNECT_RETRY_DELAY)
                 continue
-            self.log('Connected! Writing checksum ' f'file to {self.CHECKSUM_FILE}')
+            self.log(_('Connected! Writing checksum ' f'file to {self.CHECKSUM_FILE}'))
```
Comment (pandafy) [inline openwisp_firmware_upgrader/upgraders/openwrt.py:177]:
```suggestion
            self.log(_(f'Connected! Writing checksum file to {self.CHECKSUM_FILE}'))
```

**Review by pandafy — CHANGES_REQUESTED** [review]
Great progress @priyanshuone6 :+1: 

There are few minor changes requested below.

**Comment by priyanshuone6** [comment]
@pandafy I have committed all the changes you suggested, thanks a lot for reviewing. 😃

File: openwisp_firmware_upgrader/upgraders/openwrt.py [hunk 4]
```diff
@@ -90,32 +91,37 @@ def _test_checksum(self, image):
                     'Identical checksum found in the filesystem, '
                     'upgrade not needed.'
                 )
+                message = _(
+                    'Firmware already upgraded previously. '
+                    'Identical checksum found in the filesystem, '
+                    'upgrade not needed.'
+                )
```
Comment (pandafy) [inline openwisp_firmware_upgrader/upgraders/openwrt.py:98]: This is not what I meant. Can you see the problem here? `message` variable is defined twice now. The suggestions were given just to give an idea what is expected. They were never meant to be used directly.

**Review by pandafy — CHANGES_REQUESTED** [review]

File: openwisp_firmware_upgrader/upgraders/openwrt.py (hunk 4 above)
Comment (priyanshuone6) [reply openwisp_firmware_upgrader/upgraders/openwrt.py]: Sorry, I missed that, I have committed it
I am still failing some tests and I am finding it hard to solve them.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Please fix the style issues by running `openwisp-qa-format` as [described in our contributing guidelines (see the python code style conventions).](https://openwisp.io/docs/developer/contributing.html#coding-style-conventions).
Ensure also the commit message is compliant, run `./run-qa-checks` before committing.

**Review by pandafy — COMMENTED** [review]
@priyanshuone6 can you squash these commits and rebase on current master?

**Comment by atb00ker** [comment]
@priyanshuone6 this is very close to merge, could you please rebase:

```
git fetch upstream master
git rebase upstream/master
```

**Review by atb00ker — APPROVED** [review]
LGTM! :smile:
Build is failing because of an irrelevant issue.

**Review by pandafy — APPROVED** [review]
LGTM :+1: 
Nicely done @priyanshuone6!
Deferring merge to @nemesisdesign :smile: 

I will send a patch to this PR to fix the failing build.

**Review by nemesisdesign — APPROVED** [review]
Thanks

**Pull request merged**